[1m[30m-- run_tests.py: execute commands[0m[0m
[1m[30m  /usr/bin/python3 /opt/ros/noetic/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim --package=agv_sim --results-filename test_integration_test.xml --results-base-dir "/home/malcom/ros-jenkins-pipeline/ros_ws/build/agv_sim/test_results" /home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim/test/integration_test.test[0m[0m
... logging to /home/malcom/.ros/log/rostest-ubuntu-17969.log
Traceback (most recent call last):
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 297, in _arg_tag
    value, default, doc = self.opt_attrs(tag, context, ('value', 'default', 'doc'))
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 208, in opt_attrs
    return [self.resolve_args(tag_value(tag,a), context) for a in attrs]
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 208, in <listcomp>
    return [self.resolve_args(tag_value(tag,a), context) for a in attrs]
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 189, in resolve_args
    return substitution_args.resolve_args(args, context=context.resolve_dict, resolve_anon=self.resolve_anon)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/substitution_args.py", line 392, in resolve_args
    resolved = _resolve_args(resolved, context, resolve_anon, commands)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/substitution_args.py", line 405, in _resolve_args
    resolved = commands[command](resolved, a, args, context)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/substitution_args.py", line 164, in _find
    res = _find_executable(
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/substitution_args.py", line 212, in _find_executable
    full_path = _get_executable_path(rp.get_path(args[0]), path)
  File "/usr/lib/python3/dist-packages/rospkg/rospack.py", line 207, in get_path
    raise ResourceNotFound(name, ros_paths=self._ros_paths)
rospkg.common.ResourceNotFound: demo_pkg
ROS path [0]=/opt/ros/noetic/share/ros
ROS path [1]=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim
ROS path [2]=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim
ROS path [3]=/opt/ros/noetic/share

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/ros/noetic/share/rostest/cmake/../../../bin/rostest", line 71, in <module>
    rostestmain()
  File "/opt/ros/noetic/lib/python3/dist-packages/rostest/__init__.py", line 279, in rostestmain
    _main()
  File "/opt/ros/noetic/lib/python3/dist-packages/rostest/rostest_main.py", line 159, in rostestmain
    testCase = rostest.runner.createUnitTest(pkg, test_file, options.reuse_master, options.clear, options.results_base_dir)
  File "/opt/ros/noetic/lib/python3/dist-packages/rostest/runner.py", line 229, in createUnitTest
    config = roslaunch.parent.load_config_default([test_file], None)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/config.py", line 461, in load_config_default
    loader.load(f, config, argv=args, verbose=verbose)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 763, in load
    self._load_launch(launch, ros_config, is_core=core, filename=filename, argv=argv, verbose=verbose)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 735, in _load_launch
    self._recurse_load(ros_config, launch.childNodes, self.root_context, None, is_core, verbose)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 701, in _recurse_load
    val = self._include_tag(tag, context, ros_config, default_machine, is_core, verbose)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 96, in call
    return f(*args, **kwds)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 622, in _include_tag
    self._arg_tag(t, child_ns, ros_config, verbose=verbose)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 96, in call
    return f(*args, **kwds)
  File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/xmlloader.py", line 309, in _arg_tag
    raise ResourceNotFound(
rospkg.common.ResourceNotFound: The following package was not found in <arg name="world_name" value="$(find demo_pkg)/worlds/test.world"/>: demo_pkg
ROS path [0]=/opt/ros/noetic/share/ros
ROS path [1]=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim
ROS path [2]=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim
ROS path [3]=/opt/ros/noetic/share
Cannot find results, writing failure results to '/home/malcom/ros-jenkins-pipeline/ros_ws/build/agv_sim/test_results/agv_sim/MISSING-rostest-test_integration_test.xml'
[1m[30m-- run_tests.py: execute commands[0m[0m

  /usr/bin/python3 /opt/ros/noetic/share/rostest/cmake/../../../bin/rostest --pkgdir=/home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim --package=agv_sim --results-filename test_simulation_test.xml --results-base-dir "/home/malcom/ros-jenkins-pipeline/ros_ws/build/agv_sim/test_results" /home/malcom/ros-jenkins-pipeline/ros_ws/src/agv_sim/test/simulation_test.test

... logging to /home/malcom/.ros/log/rostest-ubuntu-17988.log

[ROSUNIT] Outputting test results to /home/malcom/ros-jenkins-pipeline/ros_ws/build/agv_sim/test_results/agv_sim/rostest-test_simulation_test.xml

[33m[ WARN] [1738856043.651147009]: The root link base_link has an inertia specified in the URDF, but KDL does not support a root link with an inertia.  As a workaround, you can add an extra dummy link to your URDF.[0m
[0m[ INFO] [1738856044.511243531]: Finished loading Gazebo ROS API Plugin.[0m
[0m[ INFO] [1738856044.515705499]: waitForService: Service [/gazebo/set_physics_properties] has not been advertised, waiting...[0m
[0m[ INFO] [1738856045.142814039]: waitForService: Service [/gazebo/set_physics_properties] is now available.[0m
[0m[ INFO] [1738856045.186722439, 0.014000000]: Physics dynamic reconfigure ready.[0m
[0m[ INFO] [1738856045.754053528, 0.326000000]: Laser Plugin: Using the 'robotNamespace' param: '/'[0m
[0m[ INFO] [1738856045.754191072, 0.326000000]: Starting Laser Plugin (ns = /)[0m
[0m[ INFO] [1738856045.756128338, 0.326000000]: Laser Plugin (ns = /)  <tf_prefix_>, set to ""[0m
[0m[ INFO] [1738856045.843585633, 0.326000000]: Starting plugin DiffDrive(ns = /agv/)[0m
[33m[ WARN] [1738856045.844104971, 0.326000000]: DiffDrive(ns = /agv/): missing <rosDebugLevel> default is na[0m
[0m[ INFO] [1738856045.844763383, 0.326000000]: DiffDrive(ns = /agv/): <tf_prefix> = [0m
[33m[ WARN] [1738856045.845160439, 0.326000000]: DiffDrive(ns = /agv/): missing <odometrySource> default is 1[0m
[33m[ WARN] [1738856045.845451067, 0.326000000]: GazeboRosDiffDrive Plugin (ns = ) missing <publishTf>, defaults to 1[0m
[0m[ INFO] [1738856045.846225766, 0.326000000]: DiffDrive(ns = /agv/): Advertise joint_states[0m
[0m[ INFO] [1738856045.846898904, 0.326000000]: DiffDrive(ns = /agv/): Try to subscribe to cmd_vel[0m
[0m[ INFO] [1738856045.848976098, 0.326000000]: DiffDrive(ns = /agv/): Subscribe to cmd_vel[0m
[0m[ INFO] [1738856045.850335362, 0.326000000]: DiffDrive(ns = /agv/): Advertise odom on odom [0m
[Testcase: testsimulation_test] ... ok

[ROSTEST]-----------------------------------------------------------------------

[agv_sim.rosunit-simulation_test/test_robot_movement][passed]

SUMMARY
[32m * RESULT: SUCCESS[0m
 * TESTS: 1
 * ERRORS: 0
 * FAILURES: 0

rostest log file is in /home/malcom/.ros/log/rostest-ubuntu-17988.log
